### ISSUE TITLE
Length of unit is wrong when move around the ruler (version 2.0.4) in macOS 11.1

### ISSUE THREAD

**Issue by qinfeng** [pascalpp/FreeRuler#97]
I just downloaded latest version 2.0.4 with the unit menu. However the length of unit is wrong when i move around the rule across the edge of my desktop as shown in the screen recording. I am selecting inch from the unit menu list.

![out1](https://user-images.githubusercontent.com/1988491/106409916-a7de0600-647c-11eb-88b7-d00152ecd2e6.gif)

System: macOS 11.1 on MacBook Pro (Retina, 13-inch, Late 2013)

**Comment by pascalpp** [comment]
@qinfeng do you have multiple monitors? this looks like what happens when you move the ruler to a new monitor and it recalculates the unit length using the dpi of the new monitor. You should be able to avoid this if you keep the ruler on one monitor or the other.

**Comment by qinfeng** [comment]
@pascalpp , no, i dont have multiple monitors.

**Comment by pascalpp** [comment]
Ah ok, I'll have to retest with a single display. But in the meantime, I guess just keep the left edge of the ruler on screen.

**Comment by qinfeng** [comment]
another finding is, you can see the length of unit is not the same for the vertical and horizontal rulers, even if i have chosen inch as the unit from the menu bar. Looks like the horizontal one is the right inch, the vertical is not.

**Comment by qinfeng** [comment]
And the mismatch of vertical and horizontal rulers can be corrected after the position of the rulers is reset. however, the issue is still there when i move the ruler out of the edge.

**Comment by pascalpp** [comment]
yeah looks like we have a literal edge case here. 😛

**Comment by pascalpp** [comment]
@carlfriess looks like there's a bug in how we determine which screen dpi to use for drawing ticks:
```
var screen: NSScreen? {
        guard let window = window else {
            return nil
        }
        return NSScreen.screens.first { $0.frame.contains(window.convertToScreen(frame).origin) }
    }
```

If you drag the ruler off the left or bottom edge of the screen, the above returns nil. I'll try to fix it but lemme know if you have any suggestions on how this might be improved. Maybe we can use the center point of the frame and determine which screen contains that point?

**Comment by carlfriess** [comment]
Whoops, not sure how that got through testing.. Sorry about that. Proposed a simple fix in #98.

**Comment by pascalpp** [comment]
Fixed in https://github.com/pascalpp/FreeRuler/releases/tag/v2.0.5
